src/NuGet.Core/NuGet.Build.Tasks/Common/MSBuildLogger.cs

```diff
@@ -61,7 +61,7 @@ public override void Log(ILogMessage message)
                 }
                 else
                 {
-                    var logMessage = message as IRestoreLogMessage;
+                    var logMessage = message as INuGetLogMessage;
 
                     if (logMessage == null)
                     {
```

Comment:
Line below, GH is not allowing me to comment no parts that weren't edited. 

If the message is null, then a RestoreLogMessage is created. Should be a more generic one?

Reply:
the only reason I let it stay is because I don't know why it would be null, but I was scared about breaking an existing code path. For packaging scenarios, I can guarantee it will never be null.